src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2886,7 +2886,7 @@ MergeTreeData::DataPartsVector MergeTreeData::removePartsInRangeFromWorkingSet(
     if (drop_range.min_block > drop_range.max_block)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Invalid drop range: {}", drop_range.getPartName());
 
-    auto partition_range = getDataPartsPartitionRange(drop_range.partition_id);
+    auto partition_range = getVisibleDataPartsVectorInPartition(txn, drop_range.partition_id, &lock);
```

Comment:
Fix 1: remove `Active` (or visible) parts only, do not try to remove `Outdated`, and, especially, `Deleting` parts, again